sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -452,22 +455,173 @@ def fetch_all_values(cls, cursor, counters_list, logger):
+                'availability_group:{}'.format(str(resource_group_id)),
+                'synchronization_health_desc:{}'.format(str(sync_health_desc)),
+            ]
+            metric_tags.extend(self.tags)
+            metric_name = '{}'.format(self.datadog_name)
+            selected_ag = self.cfg_instance.get('availability_group')
+
+            self.report_function(metric_name, column_val, tags=metric_tags)
+
+
+# https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-availability-replicas-transact-sql?view=sql-server-ver15
+class SqlAvailabilityReplicas(BaseSqlServerMetric):
+    TABLE = 'sys.availability_replicas'
+    DEFAULT_METRIC_TYPE = 'gauge'
+    QUERY_BASE = 'SELECT * FROM {table} as ar \
+                    inner join sys.dm_hadr_database_replica_cluster_states as dhdrcs \
+                    on ar.replica_id = dhdrcs.replica_id \
+                    inner join sys.dm_hadr_database_replica_states as dhdrs \
+                    on ar.replica_id = dhdrs.replica_id \
+                    inner join sys.availability_groups as ag \
+                    on ag.group_id = ar.group_id'.format(
+        table=TABLE
+    )
+
+    @classmethod
+    def fetch_all_values(cls, cursor, counters_list, logger):
+        return cls._fetch_generic_values(cursor, None, logger)
+
+    def fetch_metric(self, rows, columns):
+        value_column_index = columns.index(self.column)
+
+        is_primary_replica_index = columns.index('is_primary_replica')
+        failover_mode_desc_index = columns.index('failover_mode_desc')
+        replica_server_name_index = columns.index('replica_server_name')
+        resource_group_id_index = columns.index('resource_group_id')
+        is_local_index = columns.index('is_local')
+
+        for row in rows:
+            is_local = row[is_local_index]
+            resource_group_id = row[resource_group_id_index]
+            selected_ag = self.cfg_instance.get('availability_group')
+
+            if self.cfg_instance.get('only_emit_local') and not is_local:
+                continue
+
+            elif selected_ag and selected_ag != resource_group_id:
+                continue
+
+            column_val = row[value_column_index]
+            failover_mode_desc = row[failover_mode_desc_index]
+            is_primary_replica = row[is_primary_replica_index]
+            replica_server_name = row[replica_server_name_index]
+            resource_group_id = row[resource_group_id_index]
+
+            metric_tags = [
+                'replica_server_name:{}'.format(str(replica_server_name)),
+                'availability_group:{}'.format(str(resource_group_id)),
+                'is_primary_replica:{}'.format(str(is_primary_replica)),
+                'failover_mode_desc:{}'.format(str(failover_mode_desc)),
+            ]
+            metric_tags.extend(self.tags)
+            metric_name = '{}'.format(self.datadog_name)
+            selected_ag = self.cfg_instance.get('availability_group')
```

Comment:
```suggestion
```
Extra line?

Reply:
Yup originally had it there, but moved it up and forgot to delete the extra line.